tests/tests/test_tcp_teardown.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import json
+import os.path
+import tempfile
+import time
+
+from ..common_setup import standard_setup_one_client
+from ..MenderAPI import devauth, logger
+
+
+def set_long_poll_intervals(mender_device):
+    with tempfile.NamedTemporaryFile(mode="w") as tf:
+        output = mender_device.run("cat /etc/mender/mender.conf")
+        config = json.loads(output)
+        config["InventoryPollIntervalSeconds"] = 3600
+        config["UpdatePollIntervalSeconds"] = 3600
+        json.dump(config, tf)
+        tf.flush()
+        mender_device.put(
+            os.path.basename(tf.name),
+            local_path=os.path.dirname(tf.name),
+            remote_path="/etc/mender/mender.conf",
+        )
+        output = mender_device.run("cat /etc/mender/mender.conf")
+        logger.warning(output)
+
+
+def get_opened_tcp_connections(mender_device, binary_name):
+    # First probe that a process for binary_name exists
+    mender_device.run(f"pidof {binary_name}")
```

Comment:
Nit: Unused

Reply:
This is intentional (the comment above is meant to signal that it is intentional).

The point is that if `pidof something` returns empty, the whole for loop below will happily run and report `0` at the end. This command is to probe that a process with such binary name is running.

(while developing the test for example I was passing `mender-authd` instead of `mender-auth` and still the open TCP connections where "0" :smile: )